Lib/test/test_grammar.py

```diff
@@ -251,6 +251,15 @@ def check_error(test):
         check("1e3in x")
         check("1jin x")
 
+        check("0xfnot in x")
+        check("0o7not in x")
+        check("0b1not in x")
+        check("9not in x")
+        check("0not in x")
+        check("1.not in x")
+        check("1e3not in x")
+        check("1jnot in x")
```

Comment:
Maybe add a test with an underscore literal, just for completeness' sake?

```suggestion
        check("1jnot in x")
        check("1_000not in x")
```